src/nvim/insexpand.c

```diff
@@ -3291,6 +3299,34 @@ static bool get_next_completion_match(int type, ins_compl_next_state_T *st, pos_
   return found_new_match;
 }
 
+static void get_next_buffer_token(void)
+{
+  int flags = p_ic ? CP_ICASE : 0;
+  char *part;
+  char *saveptr = NULL;
+
+  FOR_ALL_BUFFERS(b) {
+    if (b->b_p_bl) {
+      char *copy = xstrdup(b->b_sfname);
+      memchrsub(copy, '\\', '/', sizeof(copy));
```

Comment:
ideally we would use some function in `path.c` to split things by the delimiters for the os. maybe `path_next_component()` can be used. Unlike `os_strtok` it won't null-terminate the string but we can always send in a computed length to `ins_compl_add()`